### PR TITLE
Add grpcio service code generation during build

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #4** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/protobufs/4" target="_blank">Graphite</a> 
    * **PR #6** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/protobufs/6" target="_blank">Graphite</a> 
      * **PR #7** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/protobufs/7" target="_blank">Graphite</a> 
        * **PR #9** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/protobufs/9" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/protobufs/9?utm_source=stack-comment).

File: mobilecoinfoundation/attestation/grpcio/build.rs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    remove_messages_from_grpcio_services(out_dir)?;
+    Ok(())
+}
+
+/// Removes the message types from the generated grpcio rust file.
+///
+/// # Arguments:
+/// * `path` - The path to the generated grpcio file.
+fn remove_messages_from_grpcio_services(
+    path: impl AsRef<Path>,
```
Comment (samdealy) [inline mobilecoinfoundation/attestation/grpcio/build.rs:30]: Do we expect this method to take anything other than a `Cow`? I think it's fine to leave as is if not, but just curious.

File: mobilecoinfoundation/attestation/grpcio/build.rs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+) -> Result<(), Box<dyn std::error::Error>> {
+    let path = path.as_ref();
+    let contents = fs::read_to_string(path)?;
+
+    // The services are appended after the message types. Each service is named
+    // `const METHOD_...`. This finds the first occurrence of that string.
```
Comment (samdealy) [inline mobilecoinfoundation/attestation/grpcio/build.rs:36]: Great comment, thanks for including.

File: mobilecoinfoundation/attestation/grpcio/build.rs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let out_dir = std::env::var("OUT_DIR")?;
+    grpcio_compiler::prost_codegen::compile_protos(
+        &["../v1/services.proto"],
+        &["../../../", "../"],
+        &out_dir,
+    )?;
+
+    // This is not ideal and is likely to be fragile longer term.
```
Comment (samdealy) [inline mobilecoinfoundation/attestation/grpcio/build.rs:16]: Agreed.

**Review by samdealy — APPROVED** [review]

File: mobilecoinfoundation/attestation/grpcio/build.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply mobilecoinfoundation/attestation/grpcio/build.rs:30]: I've been leaning on `impl AsRef<>` for the [common implementations](https://doc.rust-lang.org/std/convert/trait.AsRef.html#implementors), any `pathlike`, `stringlike`, or `[u8]` like items.
I might be overusing it, but it seems easy enough to use, to provide a more ergonomic api, even if that api is only used in one place.

**Review by awygle — APPROVED** [review]
This also looks good.

**Comment by nick-mobilecoin** [comment]
❌ This merge job failed. This pull request was not merged
Job ID: V6Mxxs7OfGV7GPJbCBGI.
See details on [graphite.dev](https://app.graphite.dev/github/pr/mobilecoinfoundation/protobufs/9)

**Pull request merged**